File: src/main/java/superstitio/cards/abstracts/AbstractLupaDamage.java
package superstitio.cards.abstracts;

import superstitio.DataManager;
import superstitio.customStrings.ModifierStringsWithSFW;
import superstitio.customStrings.HasSFWVersion;
import basemod.helpers.TooltipInfo;
import com.evacipated.cardcrawl.mod.stslib.damagemods.AbstractDamageModifier;

import java.util.ArrayList;

public abstract class AbstractLupaDamage extends AbstractDamageModifier {
    public final ModifierStringsWithSFW damageStrings;

    TooltipInfo tooltip = null;

    public AbstractLupaDamage(String id) {
        this.damageStrings = getDamageStringsWithSFW(id);
        this.automaticBindingForCards = true;
    }

    public static ModifierStringsWithSFW getDamageStringsWithSFW(String cardName) {
        return HasSFWVersion.getCustomStringsWithSFW(cardName, DataManager.modifiers, ModifierStringsWithSFW.class);
    }

    public AbstractLupaDamage removeAutoBind() {
        this.automaticBindingForCards = false;
        return this;
    }

    @Override
    public final ArrayList<TooltipInfo> getCustomTooltips() {
        if (tooltip == null) {
            tooltip = new TooltipInfo(damageStrings.getBasicInfo(), damageStrings.getDESCRIPTION());
        }
        ArrayList<TooltipInfo> tooltipInfos = new ArrayList<>();
        tooltipInfos.add(tooltip);
        return tooltipInfos;
    }

    @Override
    public final String getCardDescriptor() {
        return damageStrings.getNAME();
    }

    @Override
    public boolean isInherent() {
        return true;
    }

}

File: superstitioMain/java/superstitio/cardModifier/modifiers/AbstractLupaDamage.kt
package superstitio.cardModifier.modifiers

import basemod.helpers.TooltipInfo
import com.evacipated.cardcrawl.mod.stslib.damagemods.AbstractDamageModifier
import superstitio.DataManager
import superstitio.customStrings.interFace.StringSetUtility
import superstitio.customStrings.stringsSet.ModifierStringsSet

abstract class AbstractLupaDamage(id: String) : AbstractDamageModifier() {
    val damageStrings: ModifierStringsSet = getDamageStringsWithSFW(id)

    var tooltip: TooltipInfo? = null

    init {
        this.automaticBindingForCards = true
    }

    fun removeAutoBind(): AbstractLupaDamage {
        this.automaticBindingForCards = false
        return this
    }

    override fun getCustomTooltips(): ArrayList<TooltipInfo>? {
        if (tooltip == null) {
            tooltip = TooltipInfo(damageStrings.getBasicInfo(), damageStrings.getDESCRIPTION())
        }
        val tooltipInfos = ArrayList<TooltipInfo>()
        tooltipInfos.add(tooltip!!)
        return tooltipInfos
    }

    override fun getCardDescriptor(): String {
        return damageStrings.getNAME()
    }

    override fun isInherent(): Boolean {
        return true
    }

    companion object {
        fun getDamageStringsWithSFW(cardName: String): ModifierStringsSet {
            return StringSetUtility.getCustomStringsWithSFW(
                cardName,
                DataManager.modifiers,
                ModifierStringsSet::class.java
            )
        }
    }
}